lib/resource/Directory.js

```diff
@@ -75,7 +77,19 @@ Directory.prototype.getProvider = function getProvider(/* [options,] callback */
     return callback();
   }
 
-  return this.dataStore.getResource(this.provider.href, options, require('./Provider'), callback);
+  function providerFactory(provider) {
```

Comment:
I guess you meant to send invoke it as `providerFactory(this.provider)` instead of sending in the function itself?